pkg/queue/health/probe.go

```diff
@@ -17,18 +17,79 @@ limitations under the License.
 package health
 
 import (
+	"crypto/tls"
+	"errors"
+	"fmt"
 	"net"
+	"net/http"
 	"time"
+
+	corev1 "k8s.io/api/core/v1"
 )
 
+// HTTPProbeConfigOptions holds the HTTP probe config options
+type HTTPProbeConfigOptions struct {
+	Timeout time.Duration
+	*corev1.HTTPGetAction
+}
+
+// TCPProbeConfigOptions holds the TCP probe config options
+type TCPProbeConfigOptions struct {
+	SocketTimeout time.Duration
+	Address       string
+}
+
 // TCPProbe checks that a TCP socket to the address can be opened.
 // Did not reuse k8s.io/kubernetes/pkg/probe/tcp to not create a dependency
 // on klog.
-func TCPProbe(addr string, socketTimeout time.Duration) error {
-	conn, err := net.DialTimeout("tcp", addr, socketTimeout)
+func TCPProbe(config TCPProbeConfigOptions) error {
```

Comment:
I would also consider splitting just the changes needed for this package into its own PR.